authentik/sources/ldap/sync/vendor/freeipa.py

```diff
@@ -38,6 +38,7 @@ def check_pwd_last_set(self, attributes: dict[str, Any], user: User, created: bo
                 pwd_last_set=pwd_last_set,
             )
             user.set_unusable_password()
+            user.password_change_date = pwd_last_set
```

Comment:
the more correct solution for this would be something like

```diff
diff --git a/authentik/core/models.py b/authentik/core/models.py
index 85e8901ed..470436db0 100644
--- a/authentik/core/models.py
+++ b/authentik/core/models.py
@@ -340,6 +340,10 @@ class User(SerializerModel, GuardianUserMixin, AttributesMixin, AbstractUser):
         self.password_change_date = now()
         return super().set_password(raw_password)

+    def set_unusable_password(self):
+        self.password_change_date = now()
+        return super().set_unusable_password()
+
     def check_password(self, raw_password: str) -> bool:
         """
         Return a boolean of whether the raw_password was correct. Handles
```

instead of setting the date everywhere

Reply:
I added a combination of a) and c)